apis.md

```diff
@@ -9,3 +9,10 @@ The following OpenTracing APIs are currently available in the RFC stage:
 
 Please refer to the README files in the respective projects for examples of usage.
 
+### Implementation Notes
+
+OpenTracing APIs intend to express a similar experience to the user, while still being idiomatic to the language or framework. The following concepts apply to all variants.
+
+A **Span** is shared mutable state. It accumulates tags and logs, and can also create children. Finishing a span means capturing its state as a SpanRecord.
+
+A **SpanRecord** is an immutable representation of a completed span. It can be reported directly to an OpenTracing system, or encoded into a representation of an alternate system, such as HTrace or Zipkin.
```

Comment:
I'm slightly out of the loop on this, trying to piece together things, here's one setup that would make sense to me. Feel free to point out if I'm missing the real conversation.

`opentracing/api-python` implements the standard API in a way that makes it relatively easy to plug in "real" implementations, and provides a no-op implementation (by default I'd call it `NoopTracer`, but consistency with the other API implementations is more important here).

`openzipkin/zipkin-python` (or `openzipkin/zipkin-traces-python`, because we may want to build an query API client too and other kinds of python thingies) implements the same API, preferably by plugging into `opentracing/api-python`, making it possible for users to switch the Zipkin reporter to whatever else while keeping their application code unchanged.

For this to be possible the import path needs to not change when swapping out the backend. Two ways to do that would be stuff like `opentracing/api-django` which uses the facilities of the Django framework to do exactly this (replicate for all "supported" frameworks); or setting up light-weight classloader-like dynamic import functionality in `opentracing-api-python`. The former is more specific, while the latter is more surprising.

There's a Zipkin python reporter that should be easy to generalize (see https://github.com/prezi/django-zipkin/issues/8), with the caveat that it doesn't adhere to the standard API. This can be a good thing in that it provides an opportunity to test the plugin-like architecture, if we go that way.

Whatever approach we choose, we'll probably want to provide integration with common frameworks, especially when it comes to configuring endpoints for various reporters.